core/chains/evm/chain_set.go

```diff
@@ -386,7 +385,7 @@ func NewChainSet(opts ChainSetOpts, dbchains []types.Chain) (ChainSet, error) {
 	for i := range dbchains {
 		cid := dbchains[i].ID.String()
 		cll.logger.Infow(fmt.Sprintf("Loading chain %s", cid), "evmChainID", cid)
-		chain, err2 := newChain(dbchains[i], opts)
+		chain, err2 := newChain(dbchains[i], nodes[cid], opts)
```

Comment:
Is it possible that node may not exist for this cid?

Reply:
Yes, but that case was already handled before - the nil/empty slice was just carried around on the dbchain and plucked out via: `nodes := chain.Nodes` in `newEthClientFromChain`